File: LeetCode_258/main.cpp
/**
    LeetCode_258
    https://leetcode.com/problems/add-digits/description/
    MatrixDeity, 2017.
*/

#include <iostream>

using namespace std;

class Solution {
public:
    int addDigits(int num) {
        if (num / 10 == 0)
            return num;
        int res = 0;
        while (num != 0) {
            res += num % 10;
            num /= 10;
        }
        return addDigits(res);
    }
};

int main() {
    return 0;
}

File: LeetCode_258/main.go
/**
  LeetCode_258
  https://leetcode.com/problems/add-digits/
  MatrixDeity, 2023.
*/

package leetcode258

func addDigits(num int) int {
	if num == 0 {
		return 0
	}

	if num%9 == 0 {
		return 9
	}

	return num % 9
}
